library/core/src/alloc/layout.rs

```diff
@@ -383,7 +383,8 @@ impl Layout {
     #[inline]
     pub fn repeat_packed(&self, n: usize) -> Result<Self, LayoutError> {
         let size = self.size().checked_mul(n).ok_or(LayoutError)?;
-        Layout::from_size_align(size, self.align())
+        // The safe constructor is called here to enforce the isize size limit.
+        Layout::from_size_align(size as usize, self.align())
```

Comment:
nit: I think you can remove the the `as usize` too, so it's just a comment addition.

Reply:
... whoops, I thought I did 🙃 
```suggestion
        Layout::from_size_align(size, self.align())
```